mllib/src/main/scala/org/apache/spark/ml/util/SchemaUtils.scala

```diff
@@ -34,10 +34,11 @@ object SchemaUtils {
    * @param colName  column name
    * @param dataType  required column data type
    */
-  def checkColumnType(schema: StructType, colName: String, dataType: DataType): Unit = {
+  def checkColumnType(schema: StructType, colName: String, dataType: DataType,
+                      msg: String = ""): Unit = {
     val actualDataType = schema(colName).dataType
     require(actualDataType.equals(dataType),
-      s"Column $colName must be of type $dataType but was actually $actualDataType.")
+      s"Column $colName must be of type $dataType but was actually $actualDataType.$msg")
```

Comment:
It might be easy to make the mistake of not added a space at the beginning.  Could you please modify msg to handle that?  E.g.:
```
val message = if (msg.length > 0) " " + msg else ""
```